confidant/public/modules/history/services/archive.js

```diff
@@ -10,53 +10,148 @@
     ])
 
     .service('history.ResourceArchiveService', [
+        '$log',
         '$q',
         'credentials.archiveList',
         'blindcredentials.archiveList',
         'services.archiveList',
-        function($q, CredentialArchive, BlindCredentialArchive, ServiceArchive) {
-            function doQuery(service) {
-                var d = $q.defer(),
-                    result = service.get(function() { d.resolve(result); });
-                return d.promise;
-            }
+        function($log, $q, CredentialArchive, BlindCredentialArchive, ServiceArchive) {
             var _this = this;
-            this.resourceArchive = [];
-
-            this.updateResourceArchive = function() {
-                var credentialArchivePromise = doQuery(CredentialArchive),
-                    blindCredentialArchivePromise = doQuery(BlindCredentialArchive),
-                    serviceArchivePromise = doQuery(ServiceArchive);
-                $q.all([credentialArchivePromise, blindCredentialArchivePromise, serviceArchivePromise]).then(function(results) {
-                    var credentialArchive = results[0].credentials,
-                        blindCredentialArchive = results[1].blind_credentials,
-                        serviceArchive = results[2].services;
-                    for (var i = credentialArchive.length; i--;) {
-                        credentialArchive[i].type = 'credential';
+            this.resourceArchive = {
+                'credentials': {
+                    'service': CredentialArchive,
+                    'archive': [],
+                    'next_page': null
+                },
+                'blind_credentials': {
+                    'service': BlindCredentialArchive,
+                    'archive': [],
+                    'next_page': null
+                },
+                'services': {
+                    'service': ServiceArchive,
+                    'archive': [],
+                    'next_page': null
+                }
+            };
+            this.resourceArchiveVersion = 0;
+            this.limit = null;
+
+            function bumpVersion() {
+                _this.resourceArchiveVersion = _this.resourceArchiveVersion + 1;
+            }
+
+            function doQuery(service, page, limit) {
+                limit = limit || _this.limit;
+                page = page || null;
+                var defer = $q.defer();
+                service.get({'page': page, 'limit': limit}).$promise.then(function(result) {
+                    defer.resolve(result);
+                });
+                return defer.promise;
+            }
+
+            function mungeResourceData(data, type) {
+                data.forEach(function(resource){
+                    resource.type = type;
+                    resource.modified_date = new Date(resource.modified_date);
+                    if (type === 'services') {
```

Comment:
nit: maybe my lack of knowledge on JS, curious how easy/difficult it is to make this into a `const`.  My worry is that it's easy to mistype "services", "credentials", "blind_credentials", and would be easier to spot mis typing / errors if we used a variable.

Reply:
heh. yeah, it would be way easier to have it set as a const and I really wish it was. I'll followup later to make that happen.